backend/unit/unit_test.go

```diff
@@ -0,0 +1,207 @@
+package unit
+
+import (
+	"testing"
+	"time"
+
+	"github.com/short-d/app/mdtest"
+)
+
+type fakeDuration int64
+
+func TestParseDuration(t *testing.T) {
+	t.Parallel()
+
+	t.Run("second", func(t *testing.T) {
+		testCases := []struct {
+			name             string
+			time             string
+			expectedHasError bool
+			expectedDuration time.Duration
+			hasFakeType      bool
+			fakeDuration     fakeDuration
+		}{
+			{
+				name:             "correct format",
+				time:             "3s",
+				expectedHasError: false,
+				expectedDuration: 3 * time.Second,
+			},
+			{
+				name:             "multi digits",
+				time:             "120s",
+				expectedHasError: false,
+				expectedDuration: 120 * time.Second,
+			},
+			{
+				name:             "empty string",
+				time:             "",
+				expectedHasError: true,
+				expectedDuration: 0,
+			},
+			{
+				name:             "incorrect format",
+				time:             "3ss",
+				expectedHasError: true,
+				expectedDuration: 0,
+			},
+			{
+				name:         "custom Duration type",
+				time:         "6s",
+				hasFakeType:  true,
+				fakeDuration: fakeDuration(6 * time.Second),
+			},
```

Comment:
I think this should be part of the envconfig. The solo responsibility of unit is to parse unit, not dealing with reflection

Reply:
Should this be as part of other PR because the parsing of duration is still not added to this PR?

Reply:
@arberiii Yes, this should be part of the other PR!